docs(nm): added documentation for vlan and mtu configuration (#4919)
eclipse/kura: docs/gateway-configuration/vlan-configuration.md

# VLAN Configuration

> For devices configured to use [NetworkManager](https://networkmanager.dev), it is possible to configure multiple VLAN interfaces.

Reviewer: ```suggestion
For devices configured to use [NetworkManager](https://networkmanager.dev), it is possible to configure multiple VLAN interfaces. Take a look at our [installer profiles](../getting-started/install-kura.md#installer-types) to know which are supported.
```

This is achieved via a VLAN ID, or VLAN tag, a numerical label added to network frames to identify the specific Virtual Local Area Network (VLAN) to which they belong. It's a critical component in VLAN technology, allowing network switches and routers to differentiate and route traffic within a VLAN. VLAN tags are added to the Ethernet frame's header, indicating which virtual network a data packet should be directed to when it traverses the physical network infrastructure. Therefore, VLANs must also be supported and configured on the network equipment a device is connected to.

A VLAN can be named freely, as long as it's 15 or less characters. A typical VLAN naming format is `physicalInterfaceName`.`vlanId` (eg. a vlan with id 100 on the interface eth0 would be named eth0.100).

This is achieved by NetworkManager by creating a virtual device bound to the underlying physical interface when Kura sets up a new VLAN connection.

## VLAN Configuration via Kura Snapshot upload
Currently, VLAN configuration is supported via uploading snapshot.xml fragments.

!!! warning

>     When creating a new VLAN be sure to include the net.interfaces parameter, containing both the previously existing network interface and the name of the new vlan to be created.

Reviewer: Formatting

```suggestion
    When creating a new VLAN be sure to include the `net.interfaces` parameter, containing both the previously existing network interface and the name of the new vlan to be created.
```

                <esf:value>41</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.ip4.address" type="String">
                <esf:value>192.168.41.1</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.ip4.prefix" type="Short">
                <esf:value>24</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.vlan.flags" type="Integer">
                <esf:value>1</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.vlan.ingress" type="String">
                <esf:value>1:2,3:4</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.vlan.egress" type="String">
                <esf:value>5:6</esf:value>
            </esf:property>
        </esf:properties>
    </esf:configuration>
</esf:configurations>
```

## VLAN Management

> Once a VLAN is created it can be managed via the Kura UI just like an Ethernet interface.

Reviewer: ```suggestion
Once a VLAN is created it can be managed via the Kura UI just like any other Ethernet interface.
```

> ![VLAN UI management](./images/vlan-example.png)

Reviewer: Wrong reference resulting in broken image

![image](https://github.com/eclipse/kura/assets/22748355/3399c3c3-f462-43c3-8992-cf5b6e9dd5ec)

```suggestion

![VLAN UI management](./images/vlan-interface-example.png)
```

!!! warning

>     Setting a VLAN status to Disabled deletes its configuration in NetworkManager and is no longer visible on the UI but all the configurations are left in Kura. In can be restored by setting the `net.interface.<interface>.config.ip4.status` to `netIPv4StatusEnabledLAN` or `netIPv4StatusEnabledWAN` via snapshot upload, then resume configuration via UI.

Reviewer: Not only it deletes the configuration but the actual virtual interface. I would document that explicitly

```suggestion
    Setting a VLAN status to Disabled deletes its configuration in NetworkManager and the virtual interface from the system. Althougt it will no longer be visible on the UI, all the configurations are left in Kura. Therefore the VLAN can be restored by setting the `net.interface.<interface>.config.ip4.status` to `netIPv4StatusEnabledLAN` or `netIPv4StatusEnabledWAN` via snapshot upload, then resume configuration via UI.
```

                <esf:value>41</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.ip4.address" type="String">
                <esf:value>192.168.41.1</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.ip4.prefix" type="Short">
                <esf:value>24</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.vlan.flags" type="Integer">
                <esf:value>1</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.vlan.ingress" type="String">
                <esf:value>1:2,3:4</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.vlan.egress" type="String">
                <esf:value>5:6</esf:value>
            </esf:property>
        </esf:properties>
    </esf:configuration>
</esf:configurations>
```

## VLAN Management

> Once a VLAN is created it can be managed via the Kura UI just like any other Ethernet interface.
> ![VLAN UI management](./images/vlan-interface-example.png)

Reviewer: Spacing

```suggestion
Once a VLAN is created it can be managed via the Kura UI just like any other Ethernet interface.

![VLAN UI management](./images/vlan-interface-example.png)
```

                <esf:value>192.168.41.1</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.ip4.prefix" type="Short">
                <esf:value>24</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.vlan.flags" type="Integer">
                <esf:value>1</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.vlan.ingress" type="String">
                <esf:value>1:2,3:4</esf:value>
            </esf:property>
            <esf:property array="false" encrypted="false" name="net.interface.ens33.41.config.vlan.egress" type="String">
                <esf:value>5:6</esf:value>
            </esf:property>
        </esf:properties>
    </esf:configuration>
</esf:configurations>
```

## VLAN Management
Once a VLAN is created it can be managed via the Kura UI just like any other Ethernet interface.

![VLAN UI management](./images/vlan-interface-example.png)

!!! warning

>     Setting a VLAN status to Disabled deletes its configuration in NetworkManager and the related virtual interface from the system. Although it will is no longer be visible on the UI, all the configurations are left in Kura. Therefore the VLAN can be restored by setting the `net.interface.<interface>.config.ip4.status` to `netIPv4StatusEnabledLAN` or `netIPv4StatusEnabledWAN` via snapshot upload, then resume configuration via UI.

Reviewer: ```suggestion
    Setting a VLAN status to "_Disabled_" deletes its configuration in NetworkManager and the related virtual interface from the system. Although it will is no longer be visible on the UI, all the configurations are left in Kura. Therefore the VLAN can be restored by setting the `net.interface.<interface>.config.ip4.status` to `netIPv4StatusEnabledLAN` or `netIPv4StatusEnabledWAN` via snapshot upload, then resume configuration via UI.
```
